### DISCUSSION TITLE
Optimizing Poetry for a smaller install size

### DISCUSSION THREAD

**Discussion by alexanderankin** [python-poetry/poetry#9322] [Ideas]
Hi folks, in other languages you dont have to install the package manager (java - copy some jars, node - npm is in the node images, golang - copy a binary) - but in python we have this wonderful thing called poetry, but we have to add it to our `Dockerfile`s now.

while yes, it strictly speaking doesn't matter if you use multistage builds and just copy the venv, but installing poetry itself takes some time and bandwidth because, e.g. cleo depends on rapidfuzz, which is responsible for 14 out of the extra 49 megabytes - but maybe it would be advantageous to slim down the dependency tree for a `[minimal]` version somehow anyways?

**Comment by sbrunner** [comment]
You can see this thread https://github.com/orgs/python-poetry/discussions/1879 :-)

**Reply by alexanderankin** [reply]
yes that thread is the motivation behind even avoiding multistage in the first place, so many responses, so many ways to skin the cat, poor cats...

**Comment by maxbachmann** [comment]
https://github.com/python-poetry/cleo/issues/415 already plans to replace rapidfuzz with the built in difflib library in the next major release of cleo
